src/python/pants/backend/swift/target_types.py

```diff
@@ -0,0 +1,54 @@
+# Copyright 2022 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import annotations
+
+from pants.engine.target import (
+    COMMON_TARGET_FIELDS,
+    Dependencies,
+    MultipleSourcesField,
+    SingleSourceField,
+    Target,
+    TargetFilesGenerator,
+)
+
+SWIFT_FILE_EXTENSIONS = (".swift",)
+
+
+class SwiftSourceField(SingleSourceField):
+    expected_file_extensions = SWIFT_FILE_EXTENSIONS
+
+
+class SwiftGeneratorSourcesField(MultipleSourcesField):
+    expected_file_extensions = SWIFT_FILE_EXTENSIONS
+
+
+# -----------------------------------------------------------------------------------------------
+# `swift_source` and `swift_sources` targets
+# -----------------------------------------------------------------------------------------------
```

Comment:
I assume that the `Swift` compiler operates file-by-file, rather than directory-by-directory like `Go`'s compiler?

Reply:
Yep!